parsl/dataflow/dflow.py

```diff
@@ -218,7 +218,8 @@ def _create_task_log_info(self, task_id, fail_mode):
             task_log_info['task_fail_history'] = ",".join(self.tasks[task_id]['fail_history'])
         task_log_info['task_depends'] = None
         if self.tasks[task_id]['depends'] is not None:
-            task_log_info['task_depends'] = ",".join([str(t.tid) for t in self.tasks[task_id]['depends']])
+            task_log_info['task_depends'] = ",".join([str(t.tid) for t in self.tasks[task_id]['depends']
```

Comment:
We'll miss futures that aren't AppFuture or DataFuture here. 

What do you think about replacing with something like this : 
```
deps = ",".join([ str(t.tid) if isinstance(t, AppFuture) or isinstance(t, DataFuture) else str(t) for t in self.tasks[task_id]['depends'] ]
```

Reply:
I think that that field is used as a structured list of integers by downstream monitoring code and we should not break that by putting in a bunch of arbitrary text.

In master at the moment, you don't get dependencies recorded if you do them outside of parsl: primarily if you make your own code implement dependency ordering by waiting for `.result()` before invoking a subsequent app.

I'm fine applying that same argument here: if you do stuff outside of parsl, parsl isn't going to record it.

Reply:
I see. This is fine.